pages/api/applications/assign-reviewers.ts

```diff
@@ -0,0 +1,271 @@
+
+initializeApi();
+
+const db = firestore();
+
+const APPLICATIONS_COLLECTION = '/registrations';
+const ORGANIZERS_COLLECTION = '/member';
+
+/**
+ * Reference: https://stackoverflow.com/questions/2450954/how-to-randomize-shuffle-a-javascript-array
+ *
+ * @param array
+ *
+ */
+function shuffle(array) {
+  let currentIndex = array.length;
+
+  // While there remain elements to shuffle...
+  while (currentIndex != 0) {
+    // Pick a remaining element...
+    let randomIndex = Math.floor(Math.random() * currentIndex);
+    currentIndex--;
+
+    // And swap it with the current element.
+    [array[currentIndex], array[randomIndex]] = [array[randomIndex], array[currentIndex]];
+  }
+}
+
+/**
+ * Handles POST requests to /api/applications/assign-reviewers.
+ *
+ * Assigns reviewers to applications that do not have any reviewers assigned.
+ *
+ * @param req The HTTP request
+ * @param res The HTTP response
+ */
+async function handleAssignReviewers(req: NextApiRequest, res: NextApiResponse) {
+  // TODO: Handle user authorization
+  const {
+    query: { token },
+    headers,
+  } = req;
+
+  //
+  // Check if request header contains token
+  // TODO: Figure out how to handle the string | string[] mess.
+  const userToken = (token as string) || (headers['authorization'] as string);
+  // TODO: Extract from bearer token
+  // Probably not safe
+  const isAuthorized = await userIsAuthorized(userToken, ['super_admin']);
+
+  // TODO: enable this section
+  if (!isAuthorized) {
+    return res.status(401).send({
+      type: 'request-unauthorized',
+      message: 'Request is not authorized to perform admin functionality.',
+    });
+  }
+
+  try {
+    //  get all non-reviewed applications, no reviewer properties (collections: registration,)
+    // assumption made here: each time this api is called, it will assign 2 reviewers to all non-reviewed applications
+    const applicationsSnapshot = await db
+      .collection(APPLICATIONS_COLLECTION)
+      .where('reviewer', '==', null)
+      .get();
+
+    // Convert the QuerySnapshot to an array of application objects
+    const applications = applicationsSnapshot.docs.map((doc) => ({
+      id: doc.id,
+      ...doc.data(),
+    }));
+    // shuffle the applications to avoid bias
+    shuffle(applications);
+
+    // get all organizers (collections: member)
+    const organizersSnapshot = await db.collection(ORGANIZERS_COLLECTION).get();
+    const organizers: Organizer[] = organizersSnapshot.docs.map((doc) => {
+      const data = doc.data();
+
+      // Ensure the data has the required properties, or handle missing properties accordingly
+      return {
+        id: doc.id, // The document ID
+        name: data.name, // The name field from Firebase
+        reviewCount: data.reviewCount || 0, // The reviewCount field from Firebase
+      } as Organizer;
+    });
+
+    // create a min heap of organizers based on review count
+    const organizerHeap = new MinHeap<Organizer>(
```

Comment:
Not sure if using a min heap here will make it significantly faster than just doing linear scan on a list of organizers since we only have ~40 admins.